overcloud.yml

```diff
@@ -114,6 +114,48 @@
           parameter_defaults: "{{ parameter_defaults|default({}) | combine(nvme_params) }}"
         when: passthrough_nvme is defined
 
+      - name: set resource_registry nvme parameters
+        vars:
+          res_reg: "{{ resource_registry|default([]) }}"
+        set_fact:
+          resource_registry: "{{ res_reg + ['OS::TripleO::NodeUserData=/home/stack/firstboot-nvme.yaml'] }}"
+        when: passthrough_nvme is defined or mount_nvme is defined
+
+      - name: set tls-everywhere parameter_defaults
+        vars:
+          tls_e_params: {
+            DnsSearchDomains: ["{{ ipa_domain }}"],
+            CloudDomain: "{{ ipa_domain }}",
+            CloudName: "overcloud.{{ ipa_domain }}",
+            CloudNameInternal: "overcloud.internalapi.{{ ipa_domain }}",
+            CloudNameStorage: "overcloud.storage.{{ ipa_domain }}",
+            CloudNameStorageManagement: "overcloud.storagemgmt.{{ ipa_domain }}",
+            CloudNameCtlplane: "overcloud.ctlplane.{{ ipa_domain }}",
+            IdMServer: "{{ idm_host_fqdn }}",
+            IdMDomain: "{{ ipa_domain }}",
+            IdMInstallClientPackages: true
+          }
+        set_fact:
+          parameter_defaults: "{{ parameter_defaults|default({}) | combine(tls_e_params) }}"
+        when: tls_everywhere == true
+
+      - name: set resource_registry tls-everywhere parameters
+        vars:
+          res_reg: "{{ resource_registry|default([]) }}"
+        set_fact:
+          resource_registry: "{{ res_reg + ['OS::TripleO::Services::IpaClient=/usr/share/openstack-tripleo-heat-templates/deployment/ipa/ipaservices-baremetal-ansible.yaml'] }}"
+        when: tls_everywhere == true
+
+      - name: add templates for tls-everywhere to extra_templates
```

Comment:
--tls-everywhere options not working, is infrared not adding this templates?

Reply:
I will try using this option.

Reply:
infrared has some extra tasks with the --tls-everywhere flag that are not required in Jetpack.
This causes some errors during overcloud deployment.

```
TASK [tls-everywhere : fail] ***************************************************
task path: /home/sanjay/infrared/.venv/lib/python3.6/site-packages/infrared/common/roles/tls-everywhere/tasks/validate_overcloud.yml:10
fatal: [undercloud -> undercloud]: FAILED! => {
    "changed": false,
    "msg": "Network environment file with location /home/stack/virt/network/network-environment.yaml does not exist!"
}
```